source/Projectile.cpp

```diff
@@ -71,18 +74,30 @@ Projectile::Projectile(const Projectile &parent, const Point &offset, const Angl
 	cachedTarget = TargetPtr().get();
 	double inaccuracy = weapon->Inaccuracy();
 	if(inaccuracy)
-	{
 		this->angle += Angle::Random(inaccuracy) - Angle::Random(inaccuracy);
-		if(!parent.weapon->Acceleration())
-		{
-			// Move in this new direction at the same velocity.
-			// To maintain the sign of the velocity, Point::Length can’t be used.
-			Point referenceVector = parent.angle.Unit();
-			double parentVelocity = referenceVector.Dot(parent.velocity);
-			velocity = this->angle.Unit() * parentVelocity;
-		}
+
+	// Calculate the speed of this projectile, which is the speed of the parent
+	// plus whatever speed the submunition adds.
+	speed = parent.speed + weapon->Velocity() + Random::Real() * weapon->RandomVelocity();
```

Comment:
```suggestion
	speed = parent.velocity.Length() + weapon->Velocity() + Random::Real() * weapon->RandomVelocity();
```
Currently you only set `speed` when creating the projectile, so if the parent projectile was accelerating, then `parent.speed` is out-of-date

Reply:
Additionally recording the velocity addition/subtraction due to acceleration without turning would enable more choice for applying velocity to submunition accuracy (do we use launching velocity (as in this PR currently), or velocity due to acceleration, or both?) so I like it.  I don't know that we should replace it by updating `speed` instead.

I'll have to think about/test if that will help submunitions of turning & accelerating munitions.

Reply:
`parent.velocity` includes the velocity of the firing ship, which is exactly what we want to avoid touching. And yes, `parent.speed` is out of date on accelerating projectiles, hence why inaccuracy isn't applied in the same way to accelerating projectiles, just as was done before.

Reply:
> `parent.velocity` includes the velocity of the firing ship, which is exactly what we want to avoid touching

Oh yeah that's a good point! That's the whole purpose of the `speed` attribute.

Reply:
One could conceivably replace all the `parent.speed` references with `parent.velocity.Length()` and be able to isolate the velocity--if any--added by this submunition.

To avoid cancellation errors, it's wiser to compute the added magnitude and use that instead of `speed - parent.speed` later.

Reply:
Well, that may be but `speed` additionally carries the value through for *all layers* of submunitions.  It may work if we drop it pull from grandparent's instead, but then we also have to know when to stop.  Just sounds more complicated.

I do agree about `speed - parent.speed`.  I didn't think there was any practical reason to change it though and I think we'd need another variable so I left it.

Reply:
This change avoids the drifting behavior when firing perpendicular to the direction of traveled observed in #6695, but still retains a separate issue.
![image](https://user-images.githubusercontent.com/17688683/164371741-3282ed5e-2e24-4421-9f26-6525b844492c.png)

Firing in the direction of travel.
![image](https://user-images.githubusercontent.com/17688683/164371585-583332fb-26c8-4114-83ec-070a3cd1af0f.png)

Firing opposite the direction of travel.
![image](https://user-images.githubusercontent.com/17688683/164371624-f964f328-6a84-482d-97e8-20607386414c.png)

Reply:
Right we want to leave the frame of reference correct.  Even if we're fudging where submunitions pull their velocity within the overall weapons-fire, they certainly can't pull it from ship flight.